src/core/lib/security/credentials/tls/grpc_tls_credentials_options.cc

```diff
@@ -92,6 +92,26 @@ int grpc_tls_credentials_options_set_cert_request_type(
   return 1;
 }
 
+int grpc_tls_credentials_options_set_server_verification_option(
+    grpc_tls_credentials_options* options,
+    grpc_tls_server_verification_option server_verification_option) {
+  if (options == nullptr) {
+    gpr_log(GPR_ERROR,
+            "Invalid nullptr arguments to "
+            "grpc_tls_credentials_options_set_server_verification_option()");
+    return 0;
+  }
+  if (options->server_authorization_check_config() != nullptr &&
```

Comment:
This `if` logic is incorrect. It should be `if (server_verification_option != GRPC_TLS_SERVER_VERIFICATION && options->server_authorization_check_config() == nullptr`

Reply:
The point is that if users skip either certificate or hostname verification check, they should perform server authorization check.

Reply:
Got it. Thanks.

Reply:
Actually, TlsCredentialOptions is for both server as well as client. For server it does not make sense. Right?

Reply:
Yes you are right. It does not make sense to do the above `if` logic at the server side.